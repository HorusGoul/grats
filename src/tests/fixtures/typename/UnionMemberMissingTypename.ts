/** @gqlType */
export class User {
  /** @gqlField */
  name: string = "Alice";
}

/** @gqlType */
export class Group {
  /** @gqlField */
  name: string = "Alice Fan Club";
}

/** @gqlUnion */
export type MyUnion = User | Group;
