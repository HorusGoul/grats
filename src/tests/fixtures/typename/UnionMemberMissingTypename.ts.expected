-----------------
INPUT
----------------- 
/** @gqlType */
export class User {
  /** @gqlField */
  name: string = "Alice";
}

/** @gqlType */
export class Group {
  /** @gqlField */
  name: string = "Alice Fan Club";
}

/** @gqlUnion */
export type MyUnion = User | Group;

-----------------
OUTPUT
-----------------
src/tests/fixtures/typename/UnionMemberMissingTypename.ts:2:14 - error: Missing __typename on `User`. The type `MyUnion` is used in a union or interface, so it must have a `__typename` field.

2 export class User {
               ~~~~
src/tests/fixtures/typename/UnionMemberMissingTypename.ts:8:14 - error: Missing __typename on `Group`. The type `MyUnion` is used in a union or interface, so it must have a `__typename` field.

8 export class Group {
               ~~~~~
